### ISSUE TITLE
[Bug]: OpenAI Embedding dimensions error with Ingestion Pipeline + querying

### ISSUE THREAD

**Issue by mikelonestone** [run-llama/llama_index#10712]
### Bug Description

the new openai embedding models (like "text-embedding-3-large") which can have different dimensions are not handled by Ingestion pipeline if dimensions parameter is set to 3072 instead default 1536.

Although same embed_model is passed to both ingestion pipeline and query engine

### Version

0.10.4

### Steps to Reproduce

```python
import chromadb
from llama_index.vector_stores.chroma import ChromaVectorStore
from llama_index.embeddings.openai import OpenAIEmbedding
from llama_index.core.ingestion import IngestionPipeline
from llama_index.core.node_parser import SentenceSplitter
from llama_index.core import VectorStoreIndex
from llama_index.core.schema import Document

def demo_bug_embedding(dimensions):
    docs = [
        Document(text="AAA is blue"),
        Document(text="BBB is red")
    ]
    db = chromadb.EphemeralClient()
    chroma_collection = db.get_or_create_collection(f'main-{dimensions}')
    embed_model = OpenAIEmbedding(model="text-embedding-3-large", dimensions=dimensions)
    vector_store = ChromaVectorStore(chroma_collection=chroma_collection)

    new_pipeline = IngestionPipeline(
        transformations=[
            SentenceSplitter(chunk_size=256, chunk_overlap=0),
            embed_model
        ],
        vector_store=vector_store
    )

    new_pipeline.run(documents=docs)

    index = VectorStoreIndex.from_vector_store(vector_store)

    query_engine = index.as_query_engine(
      similarity_top_k=1,
      embeddings_model=embed_model
    )

    return query_engine.query("What is the color of AAA ?")

demo_bug_embedding(3072):
```

Here is a google colab for repro:
https://colab.research.google.com/drive/1g72E2IvBtHX-KMdg-OvtYS5I2m9R-3MN?usp=sharing

### Relevant Logs/Tracbacks

```shell
InvalidDimensionException: Embedding dimension 1536 does not match collection dimensionality 3072
```

**Comment by logan-markewich** [comment]
@mikelonestone I am not able to reproduce

```python
import chromadb
from llama_index.vector_stores.chroma import ChromaVectorStore
from llama_index.embeddings.openai import OpenAIEmbedding
from llama_index.core.ingestion import IngestionPipeline
from llama_index.core import VectorStoreIndex, Document
from llama_index.core.node_parser import SentenceSplitter

def demo_bug_embedding():
    docs = [
        Document(text="first document"),
        Document(text="second document")
    ]
    db = chromadb.EphemeralClient()
    chroma_collection = db.get_or_create_collection("main")
    vector_store = ChromaVectorStore(chroma_collection=chroma_collection)

    new_pipeline = IngestionPipeline(
        transformations=[
            SentenceSplitter(chunk_size=256, chunk_overlap=0),       
            OpenAIEmbedding(model="text-embedding-3-large", dimensions=512) #new dimensions param for text-embedding-3-large
        ],
        vector_store=vector_store
    )

    new_pipeline.run(documents=docs)

demo_bug_embedding()
```

`pip freeze | grep llama-index`
```
llama-index-core==0.10.3
llama-index-embeddings-openai==0.1.1
llama-index-vector-stores-chroma==0.1.1
```

**Comment by mikelonestone** [comment]
@logan-markewich it was in a google colab where I had restarted several time the runtime to ensure a clean env.

But I don't have the error without any change today, I guess there was still some package caching/conflict somewhere between restarts.

I still have an error with dimensions but when querying, even if both ingestion pipeline and query engine share the same embed model. 
I've made an colab example here (and updated the description of the issue):
https://colab.research.google.com/drive/1g72E2IvBtHX-KMdg-OvtYS5I2m9R-3MN?usp=sharing
